### DISCUSSION TITLE
Starting with a plant_id_eia value, how do I answer: "What type of plant is this? i.e. how does it make power?"

### DISCUSSION THREAD

**Discussion by JesseWeinstein** [catalyst-cooperative/pudl#2291] [Help Me!] [answered]
Which tables do I need to JOIN, and on what fields? Are there coding tables that are relevant?

**Comment by JesseWeinstein** [comment]
@bendnorman answered (on the OpenStreetMapUS Slack where I originally asked this):

> I would check out the **fuel_type_code_pudl** field in the **generators_eia860** table. If you’re looking for fuel type and net generation I would check out the **generation_fuel_eia923** table.

**Comment by zaneselvans** [comment, marked as answer]
Unfortunately this is complicated.

Because an EIA plant is a collection of potentially heterogeneous generation units, this question will often not have a well defined answer (at least at the plant level). How you go about answering it will depend on what aspect of the power generation you're interested in.

If you want to know what kind of generators are being used -- e.g. primary steam turbine, gas combustion turbine, heat recovering steam generator, hydraulic turbine, etc. then those are attributes of the generator, and you should probably join `generator_entities_eia` and `generators_eia860` on `plant_id_eia` and `generator_id`.

If you care more about what kind of fuel is consumed, and how efficiently it's turned into electricity, then you need to look at the generation units, which are agglomerations of boilers (which consume fuel) and generators (which make electricity). The relationships between boilers and generators can be complex. They can be many-to-many, one-to-many, or many-to-one. EIA provides some of this information, with the `unit_id_eia` identifying generators and boilers within a plant that are connected to each other, but their unit IDs are pretty incomplete.

We have inferred a much more complete association, which is available in the `boiler_generation_assn_eia860` table. The `unit_id_pudl` indicates what boilers & generators we believe to be associated with each other, and `bga_source` column indicates what method we used to infer the association. You can join the boiler and generator tables on `plant_id_eia` and either `unit_id_eia` or `unit_id_pudl` to associate boilers and generators into generation units.

Note that the associations between individual boilers and generators are also subject to change from year to year! So the `boiler_generator_assn_eia860` table also has a `report_date` column.

Because different kinds/sizes of plants & utilities are subject to different reporting requirements, information about how much fuel was consumed and what kind of fuel was consumed, and how much electricity was ultimately generated is not uniform across all plants/boilers/generators/units... but that information comes from the `generation_fuel_eia923`, `boiler_fuel_eia923` and `generation_eia923` tables. We've done a bunch of work to try and make reasonable attributions of fuel consumed and electricity generated at the generator level, but that information is currently only available in software, not in the database directly (though we are working to change that as soon as possible!) @cmgosnell knows more about that methodology.

The coding tables are relevant if you want to look up the meaning of (say) the `prime_mover_code_eia` or `energy_source_code_1` and get a longer description of what the coded value means.

**Comment by grgmiller** [comment]
As part of the [Open Grid Emissions Initiative](https://singularity.energy/open-grid-emissions#:~:text=Open%20Grid%20Emissions%20is%20an,certificate%20markets%2C%20and%20academic%20research.), we specifically answer some of these questions. For example, if you download the plant static attributes table from our data download page, we've already done the processing to identify the primary fuel type of each plant. Documentation on how we do this is [here](https://docs.singularity.energy/docs/open-grid-emissions-docs/primary_fuel-assigning-primary-fuels). Happy to answer questions about the OGE project if you're looking for pre-processed data like this.
